### ISSUE TITLE
Recent files list missing files which opened by navigation bar

### ISSUE THREAD

**Issue by ahdung** [microsoft/vscode#234958]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.95.3 (system setup)
- OS Version: Windows_NT x64 10.0.19045

Steps to Reproduce:

1. Prepare 2 files: `c:\a.txt` `c:\b.txt`
2. Click `File>Open File...` to open `c:\a.txt`, then you can see this file in `Open Recent` list, that's ok:
  ![Image](https://github.com/user-attachments/assets/5e3bee58-b04e-4c9c-b776-177000cfd6ce)
3. Click `a.txt` in navigation bar to open `c:\b.txt`:
  ![Image](https://github.com/user-attachments/assets/dafecdaf-6bde-4e6d-aeac-d9b8467af589)
4. `b.txt` will never record into `Open Recent` list, `Ctrl+R` also can't find it

**Comment by bpasero** [comment]
We only add to that list from "File" - "Open..." interactions.

**Comment by ahdung** [comment]
My scene.

One of the functions of vscode is to view logs for me, you know the log files are usually only kept for a limited time, I opened `c:\SomeApp\logs\20241202.log` today, and it was recorded, tomorrow, I want see `xxx\20241203.log`, so I can press `ctrl+r` and type `someapp` to find out the `xxx\20241202.log` and open it, and navigate to `xxx\20241203.log` by navigation bar.

I can do this every day after that, but `xxx\20241202.log` will gone one day, then I will get:
![Image](https://github.com/user-attachments/assets/ffd2c5f4-7455-4fd7-9e37-f9596f2b95fb)
Then I had to reopen `xxx\xxx.log` in windows way. 

It makes no sense not to record from navigation bar, WHY?? Is `open file ...` more noble and `navigation bar` inferior? IMO, recent files should record any opened, at least provide a option for navigation way, Or, give a chance to edit the path from `ctrl+r`.
